### ISSUE TITLE
Early Adoption the Envoy OpenSSL Version

### ISSUE THREAD

**Issue by thekief** [envoyproxy/envoy-openssl#126]
As you probably have seen by the other opened issue [1], I am interested in getting the Envoy up and running with the OpenSSL extension.
With OpenSSL 1.1.1 being EOL now, Maistra is not a feasible approach for us anymore, so we are looking into early-adopting **envoy** with the OpenSSL extension.

Me and my colleague monitored the changes you introduced over the last couple months, getting closer to a fully working server.
We have seen (and tested) that there are currently a few things open, such as the missing mapping of some BoringSSL errors to OpenSSL ones.
We are wondering, if we can help expedite the stabilisation process, to get a bit faster to a fully working release; may it be code contributions or testing, we would like to help a bit.

[1] https://github.com/envoyproxy/envoy-openssl/issues/66

**Comment by thekief** [comment]
I am not lying, telling you right now that you made my day. Do you have a an approximate timeline when we can look into it? We would not mind applying patches, etc. on my own to get things started.

**Comment by twghu** [comment]
Let's organize a chat/meet to discuss how we can collaborate.

**Comment by thekief** [comment]
Sorry for my late answer. I will contact you on your committer email, this way we will not clutter the Github issues with any unnecessary information.

**Comment by vasudev-chavan1** [comment]
Please do include me,
I will be interested in building and testing on s390x architecture.

**Comment by wufanqqfsc** [comment]
> As you probably have seen by the other opened issue [1], I am interested in getting the Envoy up and running with the OpenSSL extension. With OpenSSL 1.1.1 being EOL now, Maistra is not a feasible approach for us anymore, so we are looking into early-adopting **envoy** with the OpenSSL extension.
> 
> Me and my colleague monitored the changes you introduced over the last couple months, getting closer to a fully working server. We have seen (and tested) that there are currently a few things open, such as the missing mapping of some BoringSSL errors to OpenSSL ones. We are wondering, if we can help expedite the stabilisation process, to get a bit faster to a fully working release; may it be code contributions or testing, we would like to help a bit.
> 
> [1] #66

@thekief  do your know is this Maistra have any plan to support the openssl 3.0 ?

**Comment by thekief** [comment]
@wufanqqfsc you might swap OpenSSL 1.1.1 for OpenSSL 3.0 if you really need to continue with Maistra but as the future is this repository, I highly doubt that there will be any official support for it.

**Comment by wufanqqfsc** [comment]
@thekief  yes, that's also we want to do . May be swap OpenSSL 1.1.1 for OpenSSL 3.0 is a much better choice. there is no official release now for this project "enovoy-openssl" .
Anyway, the community of "Maistra " seems not very friendly, we have no channel to ask questions and send request.

**Comment by twghu** [comment]
Regarding OpenSSL Versions, we are only supporting OpenSSL 3.  Initially 3.0, although we expect to move to 3.1 in subsequent releases.
